hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CacheConfig.java

```diff
@@ -319,6 +337,13 @@ public boolean shouldPrefetchOnOpen() {
     return this.prefetchOnOpen;
   }
 
+  /**
+   * @return true if blocks should be cached while writing during compaction, false if not
+   */
+  public boolean shouldCacheCompactedBlocksOnWrite() {
+    return this.prefetchCompactedDataOnWrite && this.prefetchOnOpen;
```

Comment:
Oh.. So the cache on write (at compaction) happens iff prefetch config is ON ! Anyways in ur case the prefetch which is another config, is ON right?  I think this is the reason why the new config you have named that way.  But some how I feel that config name is bit misleading.

Reply:
Actually the cache size should be much bigger than the hot data set size if u want to do cache on compact.  Because the compacted away data might be already in cache (Those are flused files or a result of another compaction). Those are recently been accessed also (by the compaction thread).  This feature should be very carefully used.

Reply:
Thanks for looking at this. My understanding is that in cases where prefetch is enabled, the new file is going to be read into the cache after compaction completes anyway. So the cache size requirements are the same when this new setting is enabled. This is why I wanted to limit the scope of the cache on write to only apply where prefetching is enabled: it simply is a way to do the cache loading more efficiently while we are writing the data out rather than having to read it back after compaction is done which I've found is very expensive when data is in S3.

As far as the name goes, I struggled to come up with something intuitive - how do I explain in the name alone that this only applies when prefetching is on? I tried to convey "when prefetching, do the prefetch of compacted data on write." I'm not in love with the name and I'm open to suggestions. I didn't want to give the false impression that all compacted data is going to be cached on write. Maybe "cacheCompactedDataOnWriteIfPrefetching"? Is that too wordy?